skopt/local_solve/paramils.py

```diff
@@ -0,0 +1,109 @@
+import numpy as np
+
+from sklearn.externals.joblib import Parallel, delayed
+from sklearn.utils import check_random_state
+
+def _transform_back(x0, lower_bounds, upper_bounds):
+    return x0 * (upper_bounds - lower_bounds) + lower_bounds
+
+def _local_search(func, x0, bounds, std, n_neighbors,
+                  random_state=None, maxiter=10):
+    rng = check_random_state(random_state)
+    upper_bounds = bounds[:, 1]
+    lower_bounds = bounds[:, 0]
+
+    transformed_x_min = np.asarray(x0)
+    actual_x_min = _transform_back(
+        transformed_x_min, lower_bounds, upper_bounds)
+
+    fun_min = func(actual_x_min.reshape(1, -1))[0]
+    n_params = len(x0)
+
+    for i in range(maxiter):
+        # TODO: Cache and modify in-place.
+        transformed_points = np.tile(
+            transformed_x_min, (n_neighbors * n_params, 1))
+
+        for n_param in range(n_params):
+            curr_mean = transformed_x_min[n_param]
+            random_points = np.clip(
+                curr_mean + std * rng.randn(n_neighbors), 0, 1)
+
+            start_ptr = n_param * n_neighbors
+            end_ptr = start_ptr + n_neighbors
+            transformed_points[start_ptr: end_ptr, n_param] = random_points
+
+        actual_points = _transform_back(
+            transformed_points, lower_bounds, upper_bounds)
+        func_values = func(actual_points)
+        min_func_ind = np.argmin(func_values)
+        curr_min = func_values[min_func_ind]
+        if curr_min < fun_min:
```

Comment:
Sorry for being pedantic, but please try to chose better variable snames. E.g., here we `min_func_ind` which is associated to `curr_min` and not to `fun_min`. This is is really confusing.